src/storage/mvcc/txn.rs

```diff
@@ -551,6 +568,67 @@ impl<S: Snapshot> MvccTxn<S> {
         Ok(())
     }
 
+    /// Check the status of a transaction.
+    ///
+    /// This operation checks whether a transaction has expired it's Lock's TTL, rollback the
+    /// transaction if expired, and update the transaction's min_commit_ts according to the metadata
+    /// in the primary lock.
+    /// Returns (`lock_ttl`, `commit_ts`, `is_pessimistic_txn`).
+    /// After checking, if the lock is still alive, it retrieves the Lock's TTL; if the transaction
+    /// is committed, get the commit_ts; otherwise, if the transaction is rolled back or there's
+    /// no information about the transaction, results will be both 0.
+    pub fn check_txn_status(
+        &mut self,
+        primary_key: Key,
+        caller_start_ts: u64,
+        current_ts: u64,
+    ) -> Result<(u64, u64, bool)> {
+        match self.reader.load_lock(&primary_key)? {
+            Some(ref mut lock) if lock.ts == self.start_ts => {
+                let is_pessimistic_txn = lock.for_update_ts != 0;
+
+                if extract_physical(lock.ts) + lock.ttl < extract_physical(current_ts) {
+                    // Expired. Clean it up.
+                    if lock.lock_type == LockType::Pessimistic {
+                        self.unlock_key(primary_key);
+                    } else {
+                        self.rollback_lock(primary_key, lock)?;
+                    }
+                    MVCC_CHECK_TXN_STATUS_COUNTER_VEC.rollback.inc();
+                    return Ok((0, 0, is_pessimistic_txn));
+                }
+
+                let lock_ttl = lock.ttl;
+                // If this is a large transaction and the lock is active, push forward the minCommitTS.
+                // lock.minCommitTS == 0 may be a secondary lock, or not a large transaction.
+                if lock.min_commit_ts > 0 && caller_start_ts >= lock.min_commit_ts {
+                    lock.min_commit_ts = caller_start_ts + 1;
+
+                    if lock.min_commit_ts < current_ts {
```

Comment:
Can you give some description for `current_ts` and `caller_start_ts`? they really confused me

Reply:
I've added some description in the function's comments.

Reply:
Seems we will always meet this condition? since the current ts will always bigger than (caller_start_ts)?

Reply:
I think  `current_ts` is only used to check lock expiration. It's not related to `min_commit_ts`.

Reply:
There may be transactions whose start_ts is between `lock.ts` and `current_ts`. Update it to the `current_ts` (if it's larger) can reduce updating. @coocood